### PR TITLE
NIFI-12445: Provenance Event Listing

### PR DESCRIPTION
**Pull request by mcgilman**
NIFI-12445:
- Provenance Event Listing.
- View Provenance Event dialog.
- Provenance routing.
- Provenance search.
- Replay.
- Download content.
- View content.

### CODE REVIEW

**Comment by rfellows** [comment]
Will Review

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+  -->
+
+<h2 mat-dialog-title>Provenance Event</h2>
+<div class="provenance-event">
+    <mat-dialog-content>
+        <mat-tab-group>
+            <mat-tab label="Details">
+                <div class="tab-content py-4">
+                    <div class="absolute inset-0 flex gap-x-4">
+                        <div class="w-full">
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html:25]: we should standardize on how the spacing of these elements is done. currently it seems we are just using `mb-4` here on each child element to give some space between the pieces. I've seen `mb-5` in other places.

I would actually suggest that we use flex gapping here at the parent and not use the margin setting on each element.

```
<div class="w-full flex flex-col gap-y-2">
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+                            <div class="flex flex-col mb-4">
+                                <div>Time</div>
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html:27]: we might want to consider using `<label>` for these to be more correct.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/state/provenance-event-listing/provenance-event-listing.actions.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+export const saveProvenanceRequest = createAction(
+    '[Provenance Event Listing] Save Provenance Request',
+    props<{ request: ProvenanceRequest }>()
+);
+
+export const clearProvenanceRequest = createAction('[Provenance Event Listing] Clear Provenance Request');
+
+export const selectProvenanceEvent = createAction(
+    '[Provenance Event Listing] Select Provenance Event'
+    // props<{ request: SelectParameterContextRequest }>()
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/state/provenance-event-listing/provenance-event-listing.actions.ts:72]: can this be removed?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/ui/provenance-event-listing/provenance-event-listing.component.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            )
+            .subscribe((request) => {
+                this.store.dispatch(
+                    saveProvenanceRequest({
+                        request
+                    })
+                );
+            });
+
+        // any chances to the saved request will trigger a provenance query submission
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/ui/provenance-event-listing/provenance-event-listing.component.ts:101]:
```suggestion
        // any changes to the saved request will trigger a provenance query submission
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/ui/provenance-event-listing/provenance-event-listing.component.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+                    submitProvenanceQuery({
+                        request
+                    })
+                );
+            });
+    }
+
+    isInitialLoading(status: string): boolean {
+        return status === 'pending' || status === 'loading';
+    }
```
Comment (rfellows) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/ui/provenance-event-listing/provenance-event-listing.component.ts:132]: this name is a bit misleading. really is a `isLoading`

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html (hunk 1 above)
Comment (mcgilman) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/provenance-event-dialog/provenance-event-dialog.component.html:25]: My general strategy for spacing is to use flex gap. However, it's not always possible if some of the children are conditional. For instance, in the attributes tab in this dialog. I can however use it here. I tried `gap-y-2` as you suggested but it felt a little tight. I'll make it 3 for now and we can look to adjust more later if needed.

As for the consistency of `4` vs `5`, we should look to settle on something. What makes it a little tricky is that when used in a form with other `mat-form-fields` (which have their own padding/spacing) more or less margin may be needed.

**Review by rfellows — APPROVED** [review]
The latest changes look good, @mcgilman. Thanks

**Pull request merged**